src/sentry/incidents/models.py

```diff
@@ -498,6 +506,23 @@ def save(self, **kwargs: Any) -> None:
         self._validate_actor()
         return super().save(**kwargs)
 
+    def add_processor_hook(self, hook):
+        self._processor_hooks.append(hook)
+
+    def process_hooks(self, subscription):
+        return [func(subscription) for func in self._processor_hooks]
+
+    def clean_activated_alert(self, subscription):
+        now = timezone.now()
+        time_window = subscription.date_added + timedelta(
```

Comment:
nit: can we rename this to `window_end`?

Reply:
updated to `subscription_end` as it felt a bit more descriptive too. 🎉 for the suggestion.